framework/src/main/java/org/checkerframework/framework/util/typeinference8/types/VariableBounds.java

```diff
@@ -287,6 +287,28 @@ public void addConstraintsFromComplementaryBounds(BoundKind kind, AbstractType s
         }
       }
     }
+    if (s.isUseOfVariable()) {
+      UseOfVariable otherVar = (UseOfVariable) s;
+      switch (kind) {
+        case EQUAL:
+          otherVar.addQualifierBound(BoundKind.EQUAL, qualifierBounds.get(BoundKind.EQUAL));
+          break;
+        case LOWER:
+          otherVar.addQualifierBound(BoundKind.UPPER, qualifierBounds.get(BoundKind.EQUAL));
+          break;
+        case UPPER:
+          otherVar.addQualifierBound(BoundKind.LOWER, qualifierBounds.get(BoundKind.EQUAL));
+          break;
+      }
+
+      if (kind == BoundKind.EQUAL || kind == BoundKind.UPPER) {
```

Comment:
I think this logic would be a bit easier to read if you inlined it into the switch.  (That will lead to a touch of code duplication, but I think it's OK and it may even be fewer lines of code than presently.)